gtk/src/toga_gtk/window.py

```diff
@@ -60,6 +68,10 @@ def create(self):
     # Native event handlers
     ######################################################################
 
+    def gtk_window_state_event(self, widget, event):
+        # Get the window state flags
+        self._window_state_flags = event.new_window_state
```

Comment:
Really? There's no way to get window state flags without capturing them in an event?

Reply:
There is `GtkWindow..get_state()` but it reports outdated state and not the instantaneous state.

Reply:
> It doesn't report the last "fully activated" state, it reports the last window state and not the current window state. For example, if a window changes state from MAXIMIZED->FULLSCREEN, and then `GtkWindow.get_state()` will report MAXIMIZED.

I'm not sure I follow the distinction you're making. Isn't that exactly what I said? Or are you saying that when the state transition to FULLSCREEN is *fully completed*, the state will still report as MAXIMIZED?
 
> Also, `GtkWindow.get_state()` is not mentioned in the pygobject docs: https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Window.html and I think we should not be using an undocumented API. Moreover, I had read somewhere on the gtk docs that `GtkWindow.get_state()` was deprecated.

That's because it's [inherited from Gtk.Widget](https://docs.gtk.org/gtk3/method.Widget.get_state.html). Of course, it's a deprecated API... but then, so is *every* API in GTK that is useful...

Reply:
>Or are you saying that when the state transition to FULLSCREEN is fully completed, the state will still report as MAXIMIZED

Yes this is exactly what I mean.